src/applications/disability-benefits/686c-674/README.md

```diff
@@ -0,0 +1,71 @@
+
+## Callouts on how the front end works
+
+We used the original form system to build this application and we followed a pretty standard implementation with just a few areas worth noting.
+
+### Valid VA file number check (front end)
+
+The form is built using `/config/form.js` which is imported and used inside `/containers/App.jsx` with the `<RoutedSavableApp />` component from the platform. In `/containers/App.jsx` we dynamically render a loading indicator if the app is in an `isLoading` state from Redux. This allows us to show a loading indicator while we make the call to check if the Veteran has a valid VA file number as noted in the block quote above. The back end of that valid VA file number is detailed below in the back end section. Once the call is made and the form renders we either show the user an alert if they do not have a valid VA file number, or a `<SaveInProgressIntro />` component from the platform to start the form.
+
+### The wizard
+The first page a Veteran sees when they enter the form from the introduction page is a "wizard" where they can choose what task they want to path they want to take through the form. The original design for our form intended for us only to show the chapters that the Veteran needed to see; the idea would be that we would show the Veteran a list of tasks they could accomplish using the form and they could pick whatever ones they wanted to perform, then we would only show the Veteran the chapters they needed to see to accomplish those tasks.
+
+The way we do this in the code is by using the `depends` attribute for the optional chapters inside `/config/form.js` along with a helper method called `isChapterFieldRequired` and passing in the `formData` object and the name of that respective chapter. The `isChapterFieldRequired` function then checks if the Veteran selected the task in the wizard associated with that chapter, which we hold in the `formData["view:selectable686Options"]` array in the form data and returns `true` and shows that chapter if it corresponds to a task the Veteran selected.
+
+## The back end code
+
+There are really two separate API endpoints that we use in the 686c form, one of them dealing with the check for a valid VA file number noted in the front end sections, and one that receives the form payload on submission. The endpoint for the VA file number check is located at `vets-api/app/controllers/v0/profile/valid_va_file_numbers_controller.rb` and the endpoint for the submission payload is at `vets-api/app/controllers/v0/dependents_applications_controller.rb`.
```

Comment:
the paths to these controllers could also be links potentially.

Reply:
It makes me nervous linking to something in another repo. It seems like things could change in that repo without us knowing and these links might become 404s. Thoughts?

Reply:
I think the likelihood that these will be removed from vets-api is pretty low, given that they are vital to the functionality of the 686.

Reply:
Fair enough. Added this.